lib/api/bucketPut.js

```diff
@@ -41,7 +41,7 @@ export default function bucketPut(authInfo, request, locationConstraint, log,
     }
     // - AWS JS SDK sends a request with locationConstraint us-east-1
     // if no locationConstraint provided.
-    if (locationConstraint && config.locationConstraints &&
+    if (locationConstraint && !config.regions && config.locationConstraints &&
```

Comment:
That check had to be added because now that we have 2 configs, it's possible to have the locationConfig and the legacy version of the regular config. And if locationConfig is present, regardless of the version of the regular config, config.locationConstraints will be set.

I think there are 2 possible ways to fix this. One would be to have some logic in Config.js that would prevent config.locationConstraint from being set if the regular config was the legacy version. The other is to change the logic in conditions like this to rely on !config.region or config.restEndpoints as the indicator of legacy or new config instead of config.locationConstraint.